external_resources/tasks.py

```diff
@@ -108,3 +113,202 @@ def check_external_resources_for_breakages(self):
         return self.replace(celery.group(tasks))
 
     return None
+
+
+@app.task(bind=True)
+def submit_website_resources_to_wayback_task(self, website_name):
+    """Submit all external resources of a website to the Wayback Machine."""
+    external_resources = WebsiteContent.objects.filter(
+        website__name=website_name,
+        type=CONTENT_TYPE_EXTERNAL_RESOURCE,
+    ).select_related("external_resource_state")
+
+    tasks = [
+        submit_url_to_wayback_task.s(resource.id) for resource in external_resources
+    ]
+    if tasks:
+        return self.replace(celery.group(tasks))
+
+    return None
+
+
+@app.task(
+    bind=True,
+    acks_late=True,
+    rate_limit=WAYBACK_MACHINE_TASK_RATE_LIMIT,
+    autoretry_for=(BlockingIOError, ConnectionError, Timeout),
+    retry_backoff=True,
+    retry_backoff_max=128,
+    max_retries=7,
+)
+@single_task(7)
```

Comment:
With our current settings of a single worker for this task and a rate_limit of `0.11/s` set with this change, is it required to have an additional `redis lock`?

Reply:
The `rate_limit` is sufficient but it does not fully serve our purpose. Multiple workers will be able to pick up the task independently and that's where the `@single_task` comes in handy. It ensures only one worker processes the task at a time.

Another question could arise from this:
Could `single_task()` only suffice?

`rate_limit` divides the tasks based on the time given. And `single_task()` locks the task to a single worker for the duration specified, but it doesn't manage time-based distribution like `rate_limit` does. 

Using both together ensures efficient division of tasks across workers while making sure only one worker works on the task at the given time.